test/functional/feature_reindex_readonly.py

```diff
@@ -50,15 +51,33 @@ def reindex_readonly(self):
                     self.log.warning("Return early on Linux under root, because chattr failed.")
                     self.log.warning("This should only happen due to missing capabilities in a container.")
                     self.log.warning("Make sure to --cap-add LINUX_IMMUTABLE if you want to run this test.")
-                    return
-
-        self.log.debug("Attempt to restart and reindex the node with the unwritable block file")
-        with self.nodes[0].assert_debug_log(expected_msgs=['FlushStateToDisk', 'failed to open file'], unexpected_msgs=[]):
-            self.nodes[0].assert_start_raises_init_error(extra_args=['-reindex', '-fastprune'],
-                expected_msg="Error: A fatal internal error occurred, see debug.log for details")
+                    undo_immutable = False
+        except Exception:
+            # macOS, and *BSD
+            try:
+                subprocess.run(['chflags'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+                try:
+                    subprocess.run(['chflags', 'uchg', filename], capture_output=True, check=True)
+                    undo_immutable = subprocess.check_call(['chflags', 'nouchg', filename])
```

Comment:
```suggestion
                    undo_immutable = lambda: subprocess.check_call(['chflags', 'nouchg', filename])
```
(currently, the test is not executed on OpenBSD, as `undo_immutable` is assigned an integer of value zero, if the `chflags` call succeeds)

Reply:
damn great catch thanks, fixing